phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -5220,12 +5260,8 @@ private boolean evaluateStmtProperties(MetaProperties metaProperties, MetaProper
             throws SQLException {
         boolean changingPhoenixTableProperty = false;
 
-        if (metaProperties.getImmutableRowsProp() != null) {
+        if (metaProperties.getImmutableRowsProp() != null && table.getType() != INDEX) {
             if (metaProperties.getImmutableRowsProp().booleanValue() != table.isImmutableRows()) {
-                if (table.getImmutableStorageScheme() != ImmutableStorageScheme.ONE_CELL_PER_COLUMN) {
```

Comment:
Any change required in existing IT due to removal of this restriction?

Reply:
yes, the failing tests were due to this. updated

Reply:
Got it, saw the ignored tests. Thanks